akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -1201,15 +870,34 @@ private[remote] object ArteryTransport {
   object ShuttingDown extends RuntimeException with NoStackTrace
 
   final case class InboundStreamMatValues(
-    aeronSourceLifecycle: AeronSource.ResourceLifecycle,
+    aeronSourceLifecycle: Option[AeronSource.ResourceLifecycle],
     completed:            Future[Done])
 
-  def autoSelectPort(hostname: String): Int = {
-    val socket = DatagramChannel.open().socket()
-    socket.bind(new InetSocketAddress(hostname, 0))
-    val port = socket.getLocalPort
-    socket.close()
-    port
+  def autoSelectPort(hostname: String, udp: Boolean): Int = {
+    if (udp) {
+      val socket = DatagramChannel.open().socket()
+      socket.bind(new InetSocketAddress(hostname, 0))
+      val port = socket.getLocalPort
+      socket.close()
```

Comment:
Hmm, as we know this is not a safe way to get a port that is guaranteed to be unused on busy systems. I guess it would be hard to bind the real server to port 0 and then announce that?

Reply:
That's one of the reasons why we don't complete the startup of the actor system until it is successfully bound. I agree that it's better to use fixed ports in production. For testing it's convenient though, and in the multi-node testing we retry if it fails.

Reply:
I see, thanks.

Reply:
(I guess in common production scenarios choosing a port randomly seems also little useful.)